hadoop-hdds/docs/content/security/SecuringTDE.md

```diff
@@ -25,64 +25,80 @@ icon: lock
 
-### Setting up the Key Management Server
+Once the KMS is configured, users can create an encryption key and then create an encrypted bucket using that key. All data written to an encrypted bucket will be transparently encrypted on the server-side, and data read from the bucket will be transparently decrypted.
 
-To use TDE, admin must setup a Key Management Server and provide that URI to
-Ozone/HDFS. Since Ozone and HDFS can use the same Key Management Server, this
- configuration can be provided via *core-site.xml*.
+### Configuring TDE
 
-Property| Value
------------------------------------|-----------------------------------------
-hadoop.security.key.provider.path  | KMS uri. <br> e.g. kms://http@kms-host:9600/kms
+1.  **Set up a Key Management Server (KMS):**
+  * **Hadoop KMS:** Follow the instructions in the [Hadoop KMS documentation](https://hadoop.apache.org/docs/r3.4.1/hadoop-kms/index.html).
+  * **Ranger KMS:** Ranger KMS can also be used. For Ranger KMS, encryption keys can be managed via the Ranger KMS management console or its [REST API](https://ranger.apache.org/kms/apidocs/index.html), in addition to the `hadoop key` command line interface.
 
-### Using Transparent Data Encryption
-If this is already configured for your cluster, then you can simply proceed
-to create the encryption key and enable encrypted buckets.
+2.  **Configure Ozone:**
+    Add the following property to Ozone’s `core-site.xml`:
 
-To create an encrypted bucket, client need to:
+        <property>
+          <name>hadoop.security.key.provider.path</name>
+          <value><kms_provider_path></value>
+        </property>
 
-   * Create a bucket encryption key with hadoop key CLI, which is similar to
-  how you would use HDFS encryption zones.
+    Replace `<kms_provider_path>` with the actual URI of your KMS. For example, `kms://http@kms1.example.com:9600/kms`
 
-  ```bash
-  hadoop key create enckey
-  ```
-  The above command creates an encryption key for the bucket you want to protect.
-  Once the key is created, you can tell Ozone to use that key when you are
-  reading and writing data into a bucket.
+### Creating an Encryption Key
 
-   * Assign the encryption key to a bucket.
+Use the `hadoop key create` command to create an encryption key in the configured KMS:
 
-  ```bash
-  ozone sh bucket create -k enckey /vol/encryptedbucket
-  ```
+```shell
+  hadoop key create <key_name> [-size <key_bit_length>] [-cipher <cipher_suite>] [-description <description>]
+```
+
+* `<key_name>`: The name of the encryption key.
+* **`-size <key_bit_length>` (Optional):** Specifies the key bit length. Ozone supports **128** (default) and **256** bits.
+* **`-cipher <cipher_suite>` (Optional):** Specifies the cipher suite. Currently, only **`AES/CTR/NoPadding`** (the default) is supported.
```

Comment:
```suggestion
* `-size <key_bit_length>` (Optional): Specifies the key bit length. Ozone supports 128 (default) and 256 bits.
* `-cipher <cipher_suite>` (Optional): Specifies the cipher suite. Currently, only `AES/CTR/NoPadding` (the default) is supported.
```

Reply:
Hadoop KMS supports any number that is multiple of 8, even just 8 bits: https://github.com/apache/hadoop/blob/f1ee35b8a181581d6cc6bffe0dc5157d6d1047da/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/crypto/key/TestKeyProviderFactory.java#L157